### PR TITLE
Miscellaneous documentation TODOs

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/Attributes.md [hunk 1]
```diff
@@ -645,24 +625,8 @@ same as the current heap, or that is derived from it by heap update operations.
...
-then make the reveal lemma quantifier a layerQuantifier."
-And in that case it adds the attribute to the quantifier.
-
-There is no explicit use of the `{:layerQuantifier}` attribute
-in the Dafny tests. So I believe this attribute is only used
-internally by Dafny and not externally.
-
-TODO: Need more complete explanation of this attribute.
+
+This attribute is only used internally.
```
Comment (keyboardDrummer) [inline docs/DafnyRef/Attributes.md:629]: Why does this attribute need to be mentioned? Does it occur in attributeHelp ?

File: docs/DafnyRef/Topics.md [hunk 2]
```diff
@@ -981,7 +981,38 @@ Folks, it doesn't get any simpler than that!
...
+Dafny will issue an error message if it cannot assure itself that a variable 
+has been given a value. This may be a conservative warning: Dafny may issue an error message even if it is possible to prove, but Dafny does not, that a
+variable will always be initialized.
+
+If the type of a variable is _auto-initializable_, then a default value is used
+implicitly even if the declaration of the variable does not have an 
+explicit initializer. For example, a `bool` variable is initialized by default
+to `false` and a variable with an int-based type for which `0` is a valid value
+is auto-initialized to `0`; a non-nullable class type is not 
+not auto-initialized, but a nullable class type is auto-initalized to `null`.
```
Comment (keyboardDrummer) [inline docs/DafnyRef/Topics.md:1000]: Is the double not accidental?

**Review by keyboardDrummer — APPROVED** [review]

File: docs/DafnyRef/Attributes.md (hunk 1 above)
Comment (davidcok) [reply docs/DafnyRef/Attributes.md:629]: It does. Rustan and I are discussing removing it entirely; if so, that future PR will remove even this mention.

File: docs/DafnyRef/Topics.md (hunk 2 above)
Comment (davidcok) [reply docs/DafnyRef/Topics.md:1000]: My  typo. Now fixed.

**Review by keyboardDrummer — APPROVED** [review]

**Pull request merged**